Tests/TestingApps/PaywallsTester/PaywallsTester/Views/AppContentView.swift

```diff
@@ -139,8 +133,69 @@ struct AppContentView: View {
         }
     }
 
-    private var isPurchasesConfigured: Bool {
-        return self.customerInfoStream != nil
+    private func reconfigure(for mode: Configuration.Mode) {
+        self.configuration.reconfigure(for: mode)
+        self.observeCustomerInfoStream()
+    }
+
+    private func observeCustomerInfoStream() {
+        self.customerInfoTask?.cancel()
+        self.customerInfoTask = Task {
+            if Purchases.isConfigured {
+                for await info in Purchases.shared.customerInfoStream {
+                    self.customerInfo = info
+                    self.showingDefaultPaywall = self.showingDefaultPaywall && info.activeSubscriptions.isEmpty
+                }
+            }
+        }
```

Comment:
This is creating a retain cycle: `self` owns the task, and the task now owns `self` through this closure.
Suggested:
```suggestion
        self.customerInfoTask = Task { [weak self] in
            if Purchases.isConfigured {
                for await info in Purchases.shared.customerInfoStream {
                    guard let self else { break }
                    self.customerInfo = info
                    self.showingDefaultPaywall = self.showingDefaultPaywall && info.activeSubscriptions.isEmpty
                }
            }
        }
```

Reply:
great catch! It'll be a bigger refactor, though, 'cause weak self isn't really applicable to a struct